include/par-res-kern_shmem.h

```diff
@@ -33,3 +33,129 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <shmem.h>
 
 extern void bail_out(int);
+
+/* Handle functions first and preprocessor symbols second,  *
+ * i.e. separately, to keep the code mildly readable and    *
+ * avoid duplicating any of the function signatures, which  *
+ * could lead to errors.                                    */
+
+static void prk_shmem_init(void) {
+#if (SHMEM_MAJOR_VERSION>1) || ((SHMEM_MAJOR_VERSION == 1) && (SHMEM_MINOR_VERSION >= 2))
+    shmem_init();
+#else
+    start_pes(0);
+#endif
+}
+
+static void prk_shmem_finalize(void) {
+#if (SHMEM_MAJOR_VERSION>1) || ((SHMEM_MAJOR_VERSION == 1) && (SHMEM_MINOR_VERSION >= 2))
+    shmem_finalize();
+#else
+    /* shmem_finalize contains implicit barrier, so we include this
+     * to be consistent, even though there is no matching call to start_pes(). */
+    shmem_barrier_all();
+#endif
+}
+
+static int prk_shmem_my_pe(void) {
+#if (SHMEM_MAJOR_VERSION>1) || ((SHMEM_MAJOR_VERSION == 1) && (SHMEM_MINOR_VERSION >= 2))
+    return shmem_my_pe();
+#else
+    return _my_pe();
+#endif
+}
+
+static int prk_shmem_n_pes(void) {
+#if (SHMEM_MAJOR_VERSION>1) || ((SHMEM_MAJOR_VERSION == 1) && (SHMEM_MINOR_VERSION >= 2))
+    return shmem_my_pe();
+#else
+    return _my_pe();
+#endif
+}
+
+static void * prk_shmem_malloc(size_t n) {
+#if (SHMEM_MAJOR_VERSION>1) || ((SHMEM_MAJOR_VERSION == 1) && (SHMEM_MINOR_VERSION >= 2))
+    return shmem_malloc(n);
+#else
+    return shmalloc(n);
+#endif
+}
+
+static void * prk_shmem_realloc(void *ptr, size_t size) {
+#if (SHMEM_MAJOR_VERSION>1) || ((SHMEM_MAJOR_VERSION == 1) && (SHMEM_MINOR_VERSION >= 2))
+    return shmem_realloc(ptr,size);
+#else
+    return shrealloc(ptr,size);
+#endif
+}
+
+static void * prk_shmem_align(size_t alignment, size_t size) {
+#if (SHMEM_MAJOR_VERSION>1) || ((SHMEM_MAJOR_VERSION == 1) && (SHMEM_MINOR_VERSION >= 2))
+    return shmem_align(alignment,size);
+#else
+    return shmalign(alignment,size);
+#endif
+}
+
+static void prk_shmem_free(void * ptr) {
+#if (SHMEM_MAJOR_VERSION>1) || ((SHMEM_MAJOR_VERSION == 1) && (SHMEM_MINOR_VERSION >= 2))
+    shmem_free(ptr);
+#else
+    shfree(ptr);
+#endif
+}
+
+#if defined(SHMEM_MAJOR_VERSION) && defined(SHMEM_MINOR_VERSION)
+/* OpenSHMEM 1.3 and later */
+# if (SHMEM_MAJOR_VERSION>1) || ((SHMEM_MAJOR_VERSION == 1) && (SHMEM_MINOR_VERSION >= 2))
+#  define PRK_SHMEM_BCAST_SYNC_SIZE SHMEM_BCAST_SYNC_SIZE
+#  define PRK_SHMEM_REDUCE_SYNC_SIZE SHMEM_REDUCE_SYNC_SIZE
+#  define PRK_SHMEM_SYNC_VALUE SHMEM_SYNC_VALUE
+#  define PRK_SHMEM_REDUCE_MIN_WRKDATA_SIZE SHMEM_REDUCE_MIN_WRKDATA_SIZE
+/* OpenSHMEM 1.2 and earlier, or other SHMEM that defines VERSION symbols */
+# else
+#  define PRK_SHMEM_BCAST_SYNC_SIZE _SHMEM_BCAST_SYNC_SIZE
+#  define PRK_SHMEM_REDUCE_SYNC_SIZE _SHMEM_REDUCE_SYNC_SIZE
+#  define PRK_SHMEM_SYNC_VALUE _SHMEM_SYNC_VALUE
+#  define PRK_SHMEM_REDUCE_MIN_WRKDATA_SIZE _SHMEM_REDUCE_MIN_WRKDATA_SIZE
+# endif
+/* no VERSION symbols */
+#else
```

Comment:
I don't know of an implementation where we'll actually need this case.  The conditional should be rephrased as:

```
#ifdef HAVE_OPENSHMEM_1_2
 ... Use SHMEM_
#else
 ... Use _SHMEM_
#endif
```

That should cover any SHMEM implementation out there.  The _SHMEM_ constants have been around since the beginning of shtime.